test/helpers/story/drivers/dapp-story-test-driver.ts

```diff
@@ -174,7 +196,62 @@ export default class DappStoryTestDriver extends StoryTestDriver {
     }
   }
 
-  static async swapSushiSwap(hre: HardhatRuntimeEnvironment, loan: LoanHelpersReturn): Promise<void> {
+  static async lendAave(
+    hre: HardhatRuntimeEnvironment,
+    loan: LoanHelpersReturn
+  ): Promise<void> {
+    const { contracts } = hre
+    const { details, diamond } = loan
+    const aToken = await contracts.get<IAToken>('IAToken', {
+      at: await diamond.getAssetAToken(details.lendingToken.address),
+    })
+    const borrowedAmount = details.lendingToken.balanceOf(
+      details.borrower.address
+    )
+    await diamond
+      .connect(details.borrower.signer)
+      .aaveDeposit(
+        details.loan.id,
+        details.loan.lendingToken,
+        details.loan.borrowedAmount
+      )
+
+    const escrowAddress = await diamond.getLoanEscrow(details.loan.id)
+
+    const aDaiBalance = await aToken.balanceOf(escrowAddress)
+
+    aDaiBalance.eq(0).should.eql(false, '')
+
+    const tokenAddresses = await diamond.getEscrowTokens(details.loan.id)
+    tokenAddresses.should.include(aToken.address)
+  }
+
+  static async withdrawAave(
+    hre: HardhatRuntimeEnvironment,
+    loan: LoanHelpersReturn
+  ): Promise<void> {
+    const { contracts } = hre
+    const { details, diamond } = loan
+    const aToken = await contracts.get<IAToken>('IAToken', {
+      at: await diamond.getAssetAToken(details.lendingToken.address),
+    })
+    await diamond
+      .connect(details.borrower.signer)
+      .aaveWithdrawAll(details.loan.id, details.lendingToken.address)
+
+    const escrowAddress = await diamond.getLoanEscrow(details.loan.id)
+
+    const tokenAddresses = await diamond.getEscrowTokens(details.loan.id)
+    tokenAddresses.should.not.include(aToken.address)
+
+    const aDaiBalance = await aToken.balanceOf(escrowAddress)
+    aDaiBalance.eq(0).should.eql(true, '')
```

Comment:
again, we should test that the resulting amount withdrawn is equal to the balance of the `aToken`

Reply:
this is a withdrawAll